src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -121,4 +123,38 @@ public static Tag parseTag(String tag) throws ParseException {
         }
         return tagSet;
     }
+
+    /**
+     * Parses a {@code String prevDateMet} into a {@code PrevDateMet}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @param date The last met up date.
+     * @return PrevDateMet object that is parsed.
+     * @throws ParseException if the given {@code prevDateMet} is invalid.
+     */
+    public static PrevDateMet parsePrevDateMet(String date) throws ParseException {
+        requireNonNull(date);
+        String trimmedDate = date.trim();
+        if (!PrevDateMet.isValidPrevDateMet(trimmedDate)) {
```

Comment:
Would be better if this can be abstracted out. 
E.g. 
`boolean isDateValid = PrevDataMet.isValidPrevDateMet(trimmedDate);`

Reply:
I believe I followed the format of the other classes of AB3 and this is how they did it. So if you want to abstract this out, we would need to abstract out for Phone, Email, Name etc classes as well. 
I also feel like it is not a need since that result of isDateValid is not used again.